src/core/optimize.ts

```diff
@@ -1,43 +1,86 @@
+import { assert } from "./assert"
 import * as e from "./eval"
 import * as g from "./grad"
-import { wasmOptimizer } from "./wasmopt"
+import { collectParams } from "./params"
 import * as t from "./types"
+import { wasmOptimizer } from "./wasmopt"
 
 const useWasm = true
 
-export function optimize(
+export interface Optimizer {
+  optimize(iterations: number, observations?: Map<t.Param, number>): e.Evaluator
+}
+
+function basicOptimizer(
```

Comment:
do we need this? are there cases where we imagine that the wasm one won't be available?

Reply:
I don't think so, no. I thought it might be useful for debugging, but I'm not sure that it really is. I'll remove it.